quinn-proto/src/crypto/rustls.rs

```diff
@@ -386,10 +400,6 @@ impl crypto::PacketKey for PacketKey {
 /// satisfies this requirement.
 pub(crate) fn client_config(roots: rustls::RootCertStore) -> rustls::ClientConfig {
     let mut cfg = rustls::ClientConfig::builder()
-        .with_safe_default_cipher_suites()
-        .with_safe_default_kx_groups()
-        .with_protocol_versions(&[&rustls::version::TLS13])
-        .unwrap()
```

Comment:
Why the change here? In particular, is TLS1.3 only the default upstream now?

Reply:
IIRC rustls no longer needs a TLS 1.3 only config for QUIC to work, so there doesn't seem to be much of an advantage to doing this?

Reply:
Enabling TLS 1.2, which is strictly illegal to use in QUIC, feels pretty weird.

Reply:
It should be possible to use [`rustls::ClientConfig::builder_with_protocol_versions`](https://docs.rs/rustls/latest/rustls/struct.ConfigBuilder.html#method.with_protocol_versions) here to get the default crypto provider but explicitly specify TLS 1.3 only.